analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -67,7 +68,7 @@ public void Execute()
                     return;
                 }
                 var current = queue.Dequeue();
-                if (visited.Add(current))
+                if (visited.Add(current) && current.AddVisit() <= MaxOperationVisits)
```

Comment:
It needs to be added before the visit is done, effectively on the very next line before computing `successors`. We need to have a `State` that already has the visit registered before processing further.

The main motivation for this concise way is to keep these boring limitations out of the way as much as possible. One of the pains of the old engine is that they are spread all over the place and take a big portion of the main loop. Making the core engine logic difficult to understand.

Reply:
This has the downside that we will register visits that will not take place when we already reached the maximum.

Reply:
That is acceptable given the fact that the visits will not be used for anything else and that the program state is a dead end. So it will be only dequeued and thrown away. With no successors.